app/views/posts/show.html.erb

```diff
@@ -11,6 +11,9 @@
     <% end %>
     <h2><%= @post.title %></h2>
     <%= raw "<p>#{@post.text}</p>" %>
+    <% if can? :destroy, @post %>
+      <%= button_to "Delete", user_post_path(current_user, @post), method: :delete, data: { confirm: 'Are you sure?' } %>
```

Comment:
- You have done well so far, but I have this error, when I try to delete a post that has comments. Please fix this 🙏. A  likely reason for this is that the comments still reference the `post` being deleted. Please add `dependent: :destroy` in your `post` model. i.e 
```ruby
  belongs_to :author, class_name: 'User'
  has_many :comments, dependent: :destroy
  has_many :likes, dependent: :destroy
```

error:
![wwawera](https://user-images.githubusercontent.com/73607512/231725553-43b09ead-a474-449e-8243-ab14ef25a116.PNG)

Reply:
@mckent05 thanks